cardinal/search/filtercomponent.go

```diff
@@ -0,0 +1,118 @@
+package search
+
+import (
+	"github.com/rotisserie/eris"
+
+	"pkg.world.dev/world-engine/cardinal/types"
+	"pkg.world.dev/world-engine/cardinal/types/engine"
+)
+
+// This package involves primitives for search.
+// It involves creating and combining primitives that represent
+// filtering properties on components.
+
+type PredicateEvaluator interface {
+	Evaluate(wCtx engine.Context, id types.EntityID) (bool, error)
+}
```

Comment:
godoc here as well, would help for those who are unfamiliar with this section of the code